js/interface.js

```diff
@@ -119,6 +119,7 @@ var DynamicLists = (function() {
             _this.updateDetailsRowContainer();
             _this.renderSortColumns();
             _this.renderFilterColumns();
+            _this.updateFieldsWithColumns(dataSource.columns);
```

Comment:
Generally it's better once a variable is assigned to another variable, to use the new variable -- unless something else might be changing it before it's used.

Do any of the 4 lines above change `dataSourceColumns`? If not, then see below.

```suggestion
            _this.updateFieldsWithColumns(dataSourceColumns);
```

Reply:
@tonytlwu It seems that `dataSourceColumns` isn't changing in any of those functions. I've change the code as you've suggested.